rasa_core/policies/embedding_policy.py

```diff
@@ -0,0 +1,1022 @@
+        """Train tf graph"""
+        self.session.run(tf.global_variables_initializer())
+
+        batches_per_epoch = (len(X) // self.batch_size +
+                             int(len(X) % self.batch_size > 0))
+
+        pbar = tqdm(range(self.epochs), desc="Epochs")
+        for ep in pbar:
+            ids = np.random.permutation(X.shape[0])
+
+            ep_loss = 0
+            for i in range(batches_per_epoch):
+                start_idx = i * self.batch_size
+                end_idx = (i + 1) * self.batch_size
+
+                batch_a = X[ids[start_idx:end_idx]]
+
+                batch_pos_b = Y[ids[start_idx:end_idx]]
+                actions_for_b = actions_for_X[ids[start_idx:end_idx]]
+                # add negatives
+                batch_b = self._create_batch_b(batch_pos_b, actions_for_b)
+
+                batch_c = slots[ids[start_idx:end_idx]]
+
+                if self.nuke_slots_ones_in_epochs:
+                    if (ep + 1) % self.nuke_slots_ones_in_epochs == 0:
+                        batch_c[:] = 0
+
+                sess_out = self.session.run(
+                    {'loss': loss, 'train_op': self.train_op},
+                    feed_dict={self.a_in: batch_a,
+                               self.b_in: batch_b,
+                               self.c_in: batch_c,
+                               self.is_training: True}
+                )
+                ep_loss += sess_out.get('loss') / batches_per_epoch
+
+            pbar.set_postfix({"loss": ep_loss})
+
+            if logger.isEnabledFor(logging.INFO) and (
+                    (ep + 1) % 50 == 0 or (ep + 1) == self.epochs):
+                self._output_training_stat(X, slots,
+                                           actions_for_X, all_Y_d,
+                                           mask, ep, ep_loss)
+
+    def _output_training_stat(self,
+                              X, slots, actions_for_X, all_Y_d,
+                              mask, ep, ep_loss):
+        """Output training statistics"""
+        n = 100  # choose n examples to calculate train accuracy
+        ids = np.random.permutation(len(X))[:n]
+        all_Y_d_x = np.stack([all_Y_d for _ in range(X[ids].shape[0])])
+
+        _sim, _mask, _x = self.session.run(
+            [self.sim_op, mask, self.alignment_history],
+            feed_dict={self.a_in: X[ids],
+                       self.b_in: all_Y_d_x,
+                       self.c_in: slots[ids],
+                       self.is_training: False}
+        )
+
+        train_acc = np.sum((np.argmax(_sim, -1) ==
+                            actions_for_X[ids]) * _mask)
+        train_acc /= np.sum(_mask)
+        logger.info("epoch {} / {}: loss={:.3f}, accuracy={:.3f}"
+                    "".format((ep + 1), self.epochs,
+                              ep_loss, train_acc))
+
+        t = 9
+        for i, idx in enumerate(actions_for_X[ids[0], :t+1]):
+            print("{:.3f} -- {}".format(_x[0, t, i], self.domain.actions[idx].name()))
```

Comment:
@Ghostvv can we delete this and get rid of the domain here?

Reply:
yes i will delete it soon, I still use it for testing though